TLCGen.PluggedInItems/TLCGen.Specificator/SpecificationGenerator.cs

```diff
@@ -76,15 +40,82 @@ public static void SetDirtyFlag(WordprocessingDocument doc)
+                                    break;
+                                case SpecificationParagraphType.Header2:
+                                    body.Append(OpenXmlHelper.GetTextParagraph($"{paragraph.Text}", "Heading2", true));
+                                    break;
+                                case SpecificationParagraphType.TableHeader:
+                                    TableGenerator.UpdateTables($"{paragraph.Text}");
+                                    body.Append(OpenXmlHelper.GetTextParagraph($"Tabel {TableGenerator.NumberOfTables.ToString()}: {paragraph.Text}", styleid: "Caption"));
+                                    break;
+                                case SpecificationParagraphType.Spacer:
+                                    body.Append(OpenXmlHelper.GetTextParagraph($"{paragraph.Text}", styleid: "Footer"));
+                                    break;
+                                default:
+                                    throw new ArgumentOutOfRangeException();
+                            }
+                            break;
+
+                        case SpecificationTable table:
+                            var docTable = OpenXmlHelper.GetTable(table.TableData, firstRowVerticalText: true);
+                            body.Append(docTable);
+                            break;
+
+                        case SpecificationTable2 table:
+                            var docTable2 = OpenXmlHelper.GetTable(table.TableData2, firstRowVerticalText: false);
+                            body.Append(docTable2);
+                            break;
+
+                        case SpecificationBulletList bulletList:
+                            var docBulletList = OpenXmlHelper.GetBulletList(doc, bulletList.BulletData);
+                            body.Append(docBulletList);
+                            break;
+                    }
+                }
+            }
+        }
+
         public static void GenerateSpecification(string filename, ControllerModel c, SpecificatorDataModel model)
         {
+            var pluginData = new List<SpecificationData>();
+
+            // loopen van alle plugins
+            foreach (var plugin in TLCGenPluginManager.Default.ApplicationPlugins)
+            {
+                // check of de plugin specificatie data heeft
+                if (plugin.Item2 is ITLCGenHasSpecification specGen)
+                {
+                    var d = specGen.GetSpecificationData(c);
+                    if (d != null) pluginData.Add(d);
+                }
+            }
+
             using (var doc = WordprocessingDocument.Open(filename, true))
             {
                 // Add a main document part. 
                 var body = doc.MainDocumentPart.Document.Body;
-                body.RemoveAllChildren<Paragraph>();
-
-                //var gensWithElems = CCOLGenerator.GetAllGeneratorsWithElements(c);
+                //body.RemoveAllChildren<Paragraph>();
```

Comment:
is dit niet meer nodig? dan kan het weg toch?

Reply:
verwijderd